Track non-blocking event `whatsapp.otp.verified`

carmenlau opened the pull request:
ref #2083

louischan-oursky commented on pkg/lib/interaction/nodes/create_authenticator_whatsapp_otp.go:
```diff
@@ -29,6 +30,12 @@ func (e *EdgeCreateAuthenticatorWhatsappOTP) Instantiate(ctx *interaction.Contex
 	if err != nil {
 		return nil, err
 	}
+	err = ctx.Events.DispatchEvent(&nonblocking.WhatsappOTPVerifiedEventPayload{
```
If we add the DispatchEvent call in WhatsappCodeProvider.VerifyCode, it would be less likely we will forget to DispatchEvent in the future, when we have an additional place to call WhatsappCodeProvider.VerifyCode

carmenlau commented:
1. Moved `DispatchEvent` to `WhatsappCodeProvider.VerifyCode` in commit c853da1.
2. I rebased the `billing` branch and run `npm run gentype` in this PR (d63ac72),  the `globalTypes.generated.ts` is updated. But when I search, there are similar `AuditLogActivityType` definition in files `AuditLogEntryScreen.generated.ts`, `AuditLogScreen.generated.ts` and `UsersScreen.generated.ts` which are not updated. Is it expected, or did i miss something?

louischan-oursky commented:
You did discover a problem. The codegen config has gone through several modification https://github.com/authgear/authgear-server/pull/2081#pullrequestreview-1002577977 This modification corrected the config so that global types are placed in globalTypes.generated.ts while query- and mutation-specific types are colocated with the .graphql files. The colocated files should not contain global types. I guess when we correct the config, we did not perform `find . -name `*.generated.ts` -delete` and graphql-codegen is smart enough to only touch the part it knows. So the duplicate code is not removed when we rerun `npm run gentype`.
